src/transformers/models/pix2struct/image_processing_pix2struct.py

```diff
@@ -0,0 +1,239 @@
+
+
+if is_vision_available():
+    import PIL
+
+if is_torch_available():
+    import torch
+
+
+logger = logging.get_logger(__name__)
+
+
+class Pix2StructImageProcessor(BaseImageProcessor):
+    r"""
+    Constructs a PIX2STRUCT image processor.
+
+    Args:
+        do_resize (`bool`, *optional*, defaults to `True`):
+            Whether to resize the image's (height, width) dimensions to the specified `size`. Can be overridden by the
+            `do_resize` parameter in the `preprocess` method.
+        do_normalize (`bool`, *optional*, defaults to `True`):
+            Whether to normalize the image. Can be overridden by the `do_normalize` parameter in the `preprocess`
+            method. Can be overridden by the `do_normalize` parameter in the `preprocess` method. According to
+            Pix2Struct paper and code, the image is normalized with its own mean and standard deviation.
+        do_convert_rgb (`bool`, *optional*, defaults to `True`):
+            Whether to convert the image to RGB.
+    """
+
+    model_input_names = ["pixel_values"]
+
+    def __init__(
+        self, patch_size: Dict[str, int] = [16, 16], do_normalize: bool = True, do_convert_rgb: bool = True, **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.patch_size = patch_size
+        self.do_normalize = do_normalize
+        self.do_convert_rgb = do_convert_rgb
+
+    def extract_flattened_patches(self, image: np.ndarray, max_patches: int, **kwargs) -> np.ndarray:
+        """
+        Extract flattened patches from an image.
+
+        Args:
+            image (`np.ndarray`):
+                Image to extract flattened patches from.
+        """
+        requires_backends(self.extract_flattened_patches, "torch")
+
+        # convert to torch if
+        if not isinstance(image, torch.Tensor):
+            # check if channels first:
+            if image.shape[0] != 3:
+                image = torch.from_numpy(image).permute(2, 0, 1)
+            else:
+                image = torch.from_numpy(image)
+
+        patch_height, patch_width = self.patch_size
+        _, image_height, image_width = image.shape
+        image_height = float(image_height)
+        image_width = float(image_width)
+
+        # maximize scale s.t.
+        scale = math.sqrt(max_patches * (patch_height / image_height) * (patch_width / image_width))
+        num_feasible_rows = max(min(math.floor(scale * image_height / patch_height), max_patches), 1)
+        num_feasible_cols = max(min(math.floor(scale * image_width / patch_width), max_patches), 1)
+        resized_height = max(num_feasible_rows * patch_height, 1)
+        resized_width = max(num_feasible_cols * patch_width, 1)
+
+        image = torch.nn.functional.interpolate(
+            image.unsqueeze(0),
+            size=(resized_height, resized_width),
+            mode="bilinear",
+            align_corners=False,
+            antialias=True,
+        ).squeeze(0)
```

Comment:
Thanks! 
This seemed to introduce some rounding errors (thus slightly changing the result of `generate` in a slow test) because we need to normalize before resizing as `pillow` does not support resizing on float tensors :/ I left the commit in another branch:  43324a84ee65caf60b7a6acb0b4d33247e720cd9 , will let it like this for now and dig further